### ISSUE TITLE
Update Docs to Better Reflect the Change from requests to httpx

### ISSUE THREAD

**Issue by shoeffner** [gdcc/pyDataverse#198]
### Before we can start

Before moving on, please check some things first:

* [x] Your issue may already be reported! Please search on the [issue tracker](https://github.com/gdcc/pyDataverse/issues) before creating one.
* [x] Use our issue templates for bug reports and feature requests, if that's what you need.
* [x] Are you running the expected version of pyDataverse? (check via `pip freeze`).
* [x] Is this something you can **debug and fix**? Send a pull request! Bug fixes and documentation fixes are welcome. For more information, see the [Contributor Guide](https://pydataverse.readthedocs.io/en/master/contributing/contributing.html).
* [x] We as maintainers foster an open and welcoming environment. Be respectfull, supportive and nice to each other! :)

### Issue

This is basically a follow-up issue of #174, where requests was replaced with httpx.

There are many places in the documentation (especially types in function return values) which still document this as requests.
